lib/spack/spack/test/compilers/basics.py

```diff
@@ -737,6 +739,47 @@ def module(*args):
     assert version == test_version
 
 
+def test_get_compilers(config):
+    common = {
+        "flags": {},
+        "operating_system": "ubuntu23.10",
+        "target": "x86_64",
+        "modules": [],
+        "environment": {},
+        "extra_rpaths": [],
+    }
+    with_suffix = {
+        "spec": "gcc@13.2.0-suffix",
+        "paths": {
+            "cc": "/usr/bin/gcc-13.2.0-suffix",
+            "cxx": "/usr/bin/g++-13.2.0-suffix",
+            "f77": "/usr/bin/gfortran-13.2.0-suffix",
+            "fc": "/usr/bin/gfortran-13.2.0-suffix",
+        },
+        **common,
+    }
+    without_suffix = {
+        "spec": "gcc@13.2.0",
+        "paths": {
+            "cc": "/usr/bin/gcc-13.2.0",
+            "cxx": "/usr/bin/g++-13.2.0",
+            "f77": "/usr/bin/gfortran-13.2.0",
+            "fc": "/usr/bin/gfortran-13.2.0",
+        },
+        **common,
+    }
+
+    compilers = [{"compiler": without_suffix}, {"compiler": with_suffix}]
+
+    assert spack.compilers.get_compilers(
+        compilers, cspec=spack.spec.CompilerSpec("gcc@=13.2.0-suffix")
```

Comment:
What makes a suffix a suffix ?
I was defining "clang@14.0.5.gcc.8.3.1". Is it interpreted as a suffix or a version ? (I thought that would be a version, a weird one for sure).

Reply:
There is no concept of suffix. It's split on separators and ordered lexicographically: https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

Reply:
Yes, I remember that part of the docs, that’s how I came to the conclusion that with `clang@14.0.5.gcc.8.3.1` with was only describing a version. It’s the fact that you use a different separator and the term "suffix" that got me wondering about that.